packages/fuels-signers/src/provider.rs

```diff
@@ -144,7 +148,7 @@ impl Provider {
                 .client
                 .coins(
                     &from.hash().to_string(),
-                    None,
+                    Some(&*asset_id.to_string()),
```

Comment:
The deref is a bit confusing, you could remove it

```suggestion
                    Some(&asset_id.to_string()),
```